vdp/pipeline/v1beta/component_definition.proto

```diff
@@ -0,0 +1,388 @@
+syntax = "proto3";
+
+package vdp.pipeline.v1beta;
+
+// Google API
+import "google/api/field_behavior.proto";
+import "google/api/resource.proto";
+// Protocol Buffers Well-Known Types
+import "google/protobuf/struct.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+// VDP definitions
+import "vdp/pipeline/v1beta/common.proto";
+
+///////////////////////////////////////////////////////////////////////
+// Domain types
+///////////////////////////////////////////////////////////////////////
+
+// ComponentDefinition describes a certain type of Component.
+message ComponentDefinition {
+  // View defines how a component definition is presented.
+  enum View {
+    // Unspecified, equivalent to BASIC.
+    VIEW_UNSPECIFIED = 0;
+    // Default view, only includes basic information (removes the `spec`
+    // field).
+    VIEW_BASIC = 1;
+    // Full representation.
+    VIEW_FULL = 2;
+  }
+
+  // Defines the type of task the component will perform.
+  ComponentType type = 1 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // The component definition details.
+  oneof definition {
+    // operator definition detail
+    OperatorDefinition operator_definition = 2 [(google.api.field_behavior) = OUTPUT_ONLY];
+    // connector definition detail
+    ConnectorDefinition connector_definition = 3 [(google.api.field_behavior) = OUTPUT_ONLY];
+  }
+}
+
+// ConnectorSpec represents a specification data model.
+message ConnectorSpec {
+  // Resource specification.
+  google.protobuf.Struct resource_specification = 2 [(google.api.field_behavior) = REQUIRED];
+  // Component specification.
+  google.protobuf.Struct component_specification = 3 [(google.api.field_behavior) = REQUIRED];
+  // OpenAPI specification.
+  google.protobuf.Struct openapi_specifications = 4 [(google.api.field_behavior) = REQUIRED];
+}
+
+// ConnectorType defines the connector type based on its task features.
+enum ConnectorType {
+  // Unspecified.
+  CONNECTOR_TYPE_UNSPECIFIED = 0;
+  // Source connector.
+  CONNECTOR_TYPE_SOURCE = 1;
+  // Destination connector.
+  CONNECTOR_TYPE_DESTINATION = 2;
```

Comment:
@donch1989 can these 2 enum values be deprecated as well? I don't see them being used.

Reply:
Yes, they are deprecated, can remove them.